### ISSUE TITLE
Hide menus on downward scroll (/reveal on up-scroll)

### ISSUE THREAD

**Issue by machinaeZER0** [thunder-app/thunder#74]
Suggestion: an option (perhaps toggleable) to hide the top bar (with the hamburger menu) and bottom bar (with the home button) while scrolling through your feeds. Starting to scroll back up the feed would cause the menus to slide back into view/become accessible again.

I see a fair number of lemmy/reddit apps with this feature, and it feels like a useful thing to implement - the main benefit is that you get a not-insubstantial amount of screen real estate back as you're scrolling, which always feels like a win to me :)

**Comment by CTalvio** [comment]
Example of the feature in Relay for Reddit.
https://github.com/hjiangsu/thunder/assets/4365015/d65f7536-3d2b-4cae-870e-50c3f35d93ce

**Comment by everdred** [comment]
I understand it may be popular but I'd greatly appreciate it if this could be implemented as a toggle-able option, as one of the most refreshing things I noticed about Thunder is that it _doesn't_ have this behavior. I'm really not a fan of having basic UI elements I might want to glance at or use disappear, and there's just something fundamentally weird to me about how the unconnected act of scrolling up is how you bring back the title, filter and hamburger buttons. Maybe we can both be happy!

**Comment by Fmstrat** [comment]
For more context, and I think the OP means this, both the top and bottom bars should hide on swipe up and restore on swipe down/reload. When returning to the feed from reading a post, the bars should remain in the state they were when the item was tapped on. I also agree with making this a toggle option (default true).

**Comment by hjiangsu** [comment]
Yeah, I'll chime in on this. I think it's best suited to be a toggle!

**Comment by hjiangsu** [comment]
This is partially added in with this PR: #1029 for the top app bar. For the bottom navigation bar, there is a lot more work that would be required to get it to work well so that is on-hold for the future.

**Comment by machinaeZER0** [comment]
Heck yes!!!

**Comment by vinoff** [comment]
Oh! This is the one feature I am waiting for! Please do a release!!
